### PR TITLE
Fix compatibility wrapper for range

### PR DESCRIPTION
**Pull request by lukaszgo1**
In PR #8 I've refactored our compatibility wrapper for `range` so that it accesses  `xrange` from `__builtins__` to avoid Linter warnings under Python 3.
Unfortunately it turns out I  haven't tested this code thoroughly enough and introduced the following mistakes:
- When trying to access `xrange` I've been using indexing operator and catching `TypeError` on assumption that `__builtins__` is a dictionary under Python 2, and module under Python 3 and hence is not subscriptable when on modern Python. While this holds when in the Python interpreter (that is how I tested back then) it does not work that way when outside main module as per [this Python documentation](https://docs.python.org/2.7/reference/executionmodel.html):
> By default, when in the __main__ module, __builtins__ is the built-in module __builtin__ (note: no ‘s’); when in any other module, __builtins__ is an alias for the dictionary of the __builtin__ module itself.

In short `__builtins__` is also a dictionary under Python 3 when outside main module.
- Further to the above accessing `__builtins__` is actively discouraged by Python developers (`__builtin__` should be used instead on Python 2).

Our wrapper is fixed to take the above points into account. I've also moved it to `compat` as that is where it belongs.
Note that this is not causing any real issues as of yet, since the only case where the current code on master would fail is for NVDA running under Python 3 without `six` module available.

### CODE REVIEW

**Comment by ABuffEr** [comment]
Thanks, I'll include in the next dev build. I particularly appreciate new location under compat, it's clearly a better organized code.
